File: targets/python-web/vulns/vuln12_idor.py
from flask import Blueprint, request, jsonify
from store import get_db
bp = Blueprint("vuln12", __name__)

@bp.get("/vuln12/account")
def account():
    uid = request.args.get("id", "1")
    con = get_db()
    # VULN:VULN-12:CWE-639:logic IDOR — no authorization check tying the session to id
    row = con.execute("SELECT id, username, role FROM users WHERE id = ?", (uid,)).fetchone()
    return jsonify(dict(row) if row else {})

File: targets/jsts-web/src/vulns/vuln12_idor.ts
import { Router } from "express";
import { db } from "../store";
export const router = Router();
router.get("/vuln12/account", (req, res) => {
  // VULN:VULN-12:CWE-639:logic IDOR — no authorization check tying the session to id
  const row: any = db.prepare("SELECT id, username, role FROM users WHERE id = ?").get(String(req.query.id || "1"));
  res.json(row || {});
});
